OpenTabletDriverPlugins/WindowsInk.cs

```diff
@@ -0,0 +1,299 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using TabletDriverPlugin;
+using TabletDriverPlugin.Attributes;
+using TabletDriverPlugin.Tablet;
+using TabletDriverPlugin.Platform.Display;
+using HidSharp;
+using System;
+using System.Runtime.InteropServices;
+using TabletDriverPlugin.Logging;
+
+namespace TabletDriverPlugins
+{
+    [PluginName("Windows Ink")]
+    public class AbsoluteInk : IAbsoluteMode
+    {
+        private IEnumerable<HidDevice> Devices => DeviceList.Local.GetHidDevices();
+        private HidDevice VirtualTablet { set; get; }
+
+        private Boolean isOpen = false;
+
+        public virtual HidStream ReportStream { protected set; get; }
+
+        public void Read(IDeviceReport report)
+        {
+            if (!isOpen)
+                GetVirtualDevice();
+            if (isOpen)
+                if (report is ITabletReport tabletReport)
+                    Position(tabletReport);
+        }
+
+        public Area _displayArea, _tabletArea;
+        public IDisplay _selectedDisplay;
+        public TabletProperties _tabletProperties;
+
+        public Area Output
+        {
+            set
+            {
+                _displayArea = value;
+                UpdateCache();
+            }
+            get => _displayArea;
+        }
+
+        public Area Input
+        {
+            set
+            {
+                _tabletArea = value;
+                UpdateCache();
+            }
+            get => _tabletArea;
+        }
+
+        public IDisplay SelectedDisplay
+        {
+            set
+            {
+                _selectedDisplay = value;
+                UpdateCache();
```

Comment:
`UpdateCache()` is only used to reduce unnecessary math during the `Position()` method. No values in that method need to be refreshed so this is unnecessary.